index.src.html

```diff
@@ -1810,73 +1818,269 @@
+    2. Return the result of running the [=Clone AudioData=] algorithm with
+        [=this=].
 
+: <dfn method for=AudioData>close()</dfn>
+:: Clears all state and releases the reference to the [=media resource=].
+    Close is final.
 
-### Methods ###{#audioframe-methods}
-<dl>
-  <dt><dfn method for=AudioFrame>close()</dfn></dt>
-  <dd>
-    Immediately frees [=system resources=]. When invoked, run these steps:
-    1. Release [=system resources=] for buffer and set its value to null.
-    2. Assign `true` to the {{AudioFrame/[[detached]]}} internal slot.
+    When invoked, run these steps:
+    1. Assign `true` to the {{AudioData/[[detached]]}} internal slot.
+    2. Assign `null` to {{AudioData/[[resource reference]]}}.
+
+### Algorithms ### {#audiodata-algorithms}
+
+: <dfn>Clone AudioData</dfn> (with |data|)
+:: Run these steps:
+    1. Let |clone| be a new {{AudioData}} initialized as follows:
+        1. Let |resource| be the [=media resource=] refrenced by |data|'s
+            {{AudioData/[[resource reference]]}}.
+        2. Let |reference| be a new reference to |resource|.
+        3. Assign |reference| to {{AudioData/[[resource reference]]}}.
+        4. Assign the values of |data|'s {{AudioData/[[detached]]}},
+            {{AudioData/[[sample format]]}}, {{AudioData/[[sample rate]]}},
+            {{AudioData/[[number of frames]]}},
+            {{AudioData/[[number of channels]]}}, and
+            {{AudioData/[[timestamp]]}} slots to the corresponding slots in
+            |clone|.
+    2. Return |clone|.
+
+Audio Sample Format{#audio-sample-format}
+-----------------------------------------
+Audo sample formats describe the numeric type used to represent a single
+sample (e.g. 32-bit floating point) and the arrangement of samples from
+different channels as either interleaved or planar.
 
-    NOTE: This section needs work. We should use the name and semantics of
-        VideoFrame destroy(). Similarly, we should add clone() to make a deep
-        copy.
-  </dd>
-</dl>
+<xmp class='idl'>
+enum AudioSampleFormat {
+  "U8",
+  "S16",
+  "S32",
+  "FLT",
+  "S16P",
+  "S32P",
+  "FLTP",
+};
+</xmp>
+
+: <dfn enum-value for=AudioSampleFormat>U8</dfn>
+:: 8-bit unsigned integer samples with interleaved channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>S16</dfn>
+:: 16-bit signed integer samples with interleaved channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>S32</dfn>
+:: 32-bit signed integer samples with interleaved channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>FLT</dfn>
+:: 32-bit float samples with interleaved channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>S16P</dfn>
+:: 16-bit signed integer samples with planar channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>S32P</dfn>
+:: 32-bit signed integer samples with planar channel arrangement.
+
+: <dfn enum-value for=AudioSampleFormat>FLTP</dfn>
+:: 32-bit float samples with planar channel arrangement.
```

Comment:
We'll need to explain what interleaved and planar are. Also how does those data types hold audio: for example, 16-bit is linear audio from in `[-65536, 65536]`, but the `FLT` variants only use [-1.0, 1.0] in general (but it can easily go outside this). And also `S32` is only expected to use 24-bits probably?